pyOCD/target/cortex_m.py

```diff
@@ -488,7 +488,10 @@ def writeMemory(self, addr, value, transfer_size = 32):
         write a memory location.
         By default the transfer size is a word
         """
-        self.transport.writeMem(addr, value, transfer_size)
+        try:
+            self.transport.writeMem(addr, value, transfer_size)
+        except:
```

Comment:
You would want to check at most or every location that throws a ValueError.  That's probably outside the scope this patch, so it might be easiest to just catch ValueError (This will catch TransferErrors as well since it inherits from ValueError).